### DISCUSSION TITLE
[MSW] Warning: intercepted a request without a matching request handler:

### DISCUSSION THREAD

**Discussion by Joseamica** [epicweb-dev/epic-stack#552] [Q&A]
Im using this stack for my main project, but when I use fetch I get a bunch of warnings, around 10x this message:
```
Read more: https://mswjs.io/docs/getting-started/mocks
GET /dashboard/tables?_data=routes/dashboard+/tables 200 - - 118.309 ms
GET /dashboard/tables?_data=root 200 - - 6.871 ms
[MSW] Warning: intercepted a request without a matching request handler:

  • GET https://av.fly.dev/api/clmp0brnr000dq39v4379wyp0/tables

If you still wish to intercept this unhandled request, please create a request handler for it.
Read more: https://mswjs.io/docs/getting-started/mocks
GET /dashboard/tables?_data=routes/dashboard+/tables 200 - - 91.385 ms
[MSW] Warning: intercepted a request without a matching request handler:

  • GET https://av.fly.dev/api/clmp0brnr000dq39v4379wyp0/tables

If you still wish to intercept this unhandled request, please create a request handler for it.
Read more: https://mswjs.io/docs/getting-started/mocks
GET /dashboard/tables?_data=root 200 - - 12.040 ms
GET /dashboard/tables?_data=routes/dashboard+/tables 200 - - 108.198 ms
GET /dashboard/tables?_data=root 200 - - 12.283 ms
[MSW] Warning: intercepted a request without a matching request handler:

  • GET https://av.fly.dev/api/clmp0brnr000dq39v4379wyp0/tables

If you still wish to intercept this unhandled request, please create a request handler for it.
Read more: https://mswjs.io/docs/getting-started/mocks
GET /dashboard/tables?_data=routes/dashboard+/tables 200 - - 68.092 ms
```

Also, when i just reload the app i get a lot of get requests
```
GET / 200 - - 33.989 ms
GET /?_data=root 200 - - 17.430 ms
GET /?_data=root 200 - - 2.206 ms
GET /?_data=root 200 - - 3.146 ms

```

**Comment by ananni13** [comment]
The warnings come from MSW, the mock server that is launched alongside the app server during development (see [server/dev-server.js](https://github.com/epicweb-dev/epic-stack/blob/main/server/dev-server.js#L13), [index.js](https://github.com/epicweb-dev/epic-stack/blob/main/index.js#L32-L34) and [tests/mocks/index.ts](https://github.com/epicweb-dev/epic-stack/blob/main/tests/mocks/index.ts)).

It intercepts all outgoing requests, and since you're making GET requests to `https://av.fly.dev` and you haven't defined a mock handler for that endpoint, you get the warning.

You can take a look in the [mocks folder](https://github.com/epicweb-dev/epic-stack/tree/main/tests/mocks) for examples on how to define handlers.

> Also, when i just reload the app i get a lot of get requests

> GET / 200 - - 33.989 ms
> GET /?_data=root 200 - - 17.430 ms
> GET /?_data=root 200 - - 2.206 ms
> GET /?_data=root 200 - - 3.146 ms

This is Remix calling loaders and revalidating data, multiple requests might be related to client-hints cookies:

- https://github.com/epicweb-dev/epic-stack/blob/main/docs/decisions/005-client-pref-cookies.md
- https://github.com/epicweb-dev/epic-stack/blob/main/docs/client-hints.md
- https://github.com/epicweb-dev/client-hints

**Reply by Joseamica** [reply]
Thanks for the reply!

Sorry for my ignorance, but its normal that a lot of get requests get called? Or i should take a look to see why its happening

**Comment by adamreisnz** [comment]
For anyone else running into this, you can set up a passthrough handler at the end of your handlers list to avoid warnings about unhandled requests:

```
import {http, passthrough} from 'msw'

export const handlers = [
  //...your handlers
  
  //Pass through for any other requests
  http.all('*', () => passthrough()),
]
```
